### ISSUE TITLE
'vmware-iso' errored: Script exited with non-zero exit status: 36

### ISSUE THREAD

**Issue by z3r0-D7** [rapid7/metasploitable3#476]
## Issue Description
I am trying to install metasploitable3-win2k8 on my macbook using VMware Fusion and installation breaks/exits at the same stage, pointing very likely to the glassfish service. 
I have tried this multiple times and have increased the timeout of the relevant glassfish install scripts to 30s as seen in the output below, based on discussions from a [similar issue](https://github.com/rapid7/metasploitable3/issues/155):
```
$ grep Start *glass*
setup_glassfish.bat:powershell -Command "Start-Sleep -s 30"
start_glassfish_service.bat:powershell -command "Start-Sleep -s 30"
```
This is likely related to the glassfish service issue noted on the [General Issues]( https://github.com/rapid7/metasploitable3/wiki/General-Issues) page - but the workaround for this was to downgrade to VMware Fusion v9, which is no longer available for download on the vendor site, and would be quite dated compared to the current version of v11.5.5. 

Please advise on the best path forward here. Is there a plan to fix  this issue, and what would be the workaround on current VMware versions. 

## System
* OS: MacOS Catalina version 10.15.6
* Packer Version: 1.6.1
* Vagrant Version: 2.2.9
* VirtualBox Version: N/A
* VMware Fusion Version: Professional Version 11.5.5 

## Command Output

>     [....]
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>mkdir C:\glassfish
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>powershell -Command "[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; (New-Object System.Net.WebClient).DownloadFile('http://download.java.net/glassfish/4.0/release/glassfish-4.0.zip', 'C:\Windows\Temp\glassfish4.zip')"  0<NUL
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>cmd /c ""C:\Program Files\7-Zip\7z.exe" -y x "C:\Windows\Temp\glassfish4.zip" -oC:\glassfish"
>     vmware-iso:
>     vmware-iso: 7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21
>     vmware-iso:
>     vmware-iso: Scanning the drive for archives:
>     vmware-iso: 1 file, 102178360 bytes (98 MiB)
>     vmware-iso:
>     vmware-iso: Extracting archive: C:\Windows\Temp\glassfish4.zip
>     vmware-iso: --
>     vmware-iso: Path = C:\Windows\Temp\glassfish4.zip
>     vmware-iso: Type = zip
>     vmware-iso: Physical Size = 102178360
>     vmware-iso:
>     vmware-iso: Everything is Ok
>     vmware-iso:
>     vmware-iso: Folders: 364
>     vmware-iso: Files: 1582
>     vmware-iso: Size:       124661519
>     vmware-iso: Compressed: 102178360
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>copy /Y "C:\vagrant\resources\glassfish\admin-keyfile" "C:\glassfish\glassfish4\glassfish\domains\domain1\config\admin-keyfile"
>     vmware-iso:         1 file(s) copied.
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>copy /Y "C:\vagrant\resources\glassfish\domain.xml" "C:\glassfish\glassfish4\glassfish\domains\domain1\config\domain.xml"
>     vmware-iso:         1 file(s) copied.
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>cmd.exe /c "C:\glassfish\glassfish4\bin\asadmin.bat create-service domain1"
> ==> vmware-iso: Error: loading: C:\Program Files\Java\jre1.8.0_261\bin\vcruntime140.dll
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>net start domain1
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>powershell -Command "Start-Sleep -s 30"
> ==> vmware-iso: Error: loading: C:\Program Files\Java\jre1.8.0_261\bin\server\jvm.dll
> ==> vmware-iso: The service name is invalid.
> ==> vmware-iso:
> ==> vmware-iso: More help is available by typing NET HELPMSG 2185.
> ==> vmware-iso:
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>net stop domain1
> ==> vmware-iso: The service name is invalid.
> ==> vmware-iso:
>     vmware-iso:
> ==> vmware-iso: More help is available by typing NET HELPMSG 2185.
>     vmware-iso: C:\Users\vagrant>icacls "C:\glassfish" /grant "NT Authority\LOCAL SERVICE:(OI)(CI)F" /T
> ==> vmware-iso:
>     vmware-iso: processed file: C:\glassfish
>     vmware-iso: processed file: C:\glassfish\glassfish4
> 
> [-- followed by multiple log lines --]
> 
>    vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\Image.FmriState.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\Image.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\Image.ImagePlan.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\ImagePlanProgressTracker.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\LicenseAction.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\LinkAction.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\Manifest.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\package-frame.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\package-summary.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\package-tree.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\SetAction.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\SystemInfo.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\SystemInfo.UpdateCheckFrequency.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\Variant.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\Version.DotSequence.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\com\sun\pkg\client\Version.html
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\javadocs\resources\inherit.gif
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\lib\pkg-bootstrap.jar
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\lib\pkg-bootstub.bat
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\lib\pkg-bootstub.sh
>     vmware-iso: processed file: C:\glassfish\glassfish4\pkg\lib\pkg-client.jar
>     vmware-iso: Successfully processed 1947 files; Failed processing 0 files
>     vmware-iso:
>     vmware-iso: C:\Users\vagrant>sc config "domain1" obj= "NT Authority\LOCAL SERVICE"
>     vmware-iso: [SC] OpenService FAILED 1060:
>     vmware-iso:
>     vmware-iso: The specified service does not exist as an installed service.
>     vmware-iso:
> ==> vmware-iso: Provisioning step had errors: Running the cleanup provisioner, if present...
> ==> vmware-iso: Stopping virtual machine...
> ==> vmware-iso: Deleting output directory...
> Build 'vmware-iso' errored: Script exited with non-zero exit status: 36.Allowed exit codes are: [0]
> 
> ==> Some builds didn't complete successfully and had errors:
> --> vmware-iso: Script exited with non-zero exit status: 36.Allowed exit codes are: [0]
> 
> ==> Builds finished but no artifacts were created.
>

**Comment by AaronCamp** [comment]
I have the same problem, but I'm building it for VirtualBox under Ubuntu.

**Comment by z3r0-D7** [comment]
Just to update on this issue for anyone else who runs into it. 
From further troubleshooting I found the line below to be the root cause of the issue:
```
822     vmware-iso: C:\Users\vagrant>cmd.exe /c "C:\glassfish\glassfish4\bin\asadmin.bat create-service domain1"
823 ==> vmware-iso: Error: loading: C:\Program Files\Java\jre1.8.0_261\bin\vcruntime140.dll
824 ==> vmware-iso: Error: loading: C:\Program Files\Java\jre1.8.0_261\bin\server\jvm.dll
```

In order to be able to proceed with my install, I commented out the erring parts of the related install script (setup_glassfish.bat):
```
$ pwd
../metasploitable3-workspace/metasploitable3/scripts/installs

$ grep REM setup_glassfish.bat
REM cmd.exe /c "C:\glassfish\glassfish4\bin\asadmin.bat create-service domain1"
REM net start domain1
REM powershell -Command "Start-Sleep -s 30"
REM net stop domain1
REM icacls "C:\glassfish" /grant "NT Authority\LOCAL SERVICE:(OI)(CI)F" /T
REM sc config "domain1" obj= "NT Authority\LOCAL SERVICE"
```

This allowed the installation to complete.
Even though from the install log output, jre1.8.0_261 was meant to have been installed:
```
vmware-iso: C:\Windows\Temp\chocolatey\jre8\8.0.2610.12\JRE8x64.exe
vmware-iso: Installing JRE 8.0.2610.12 64-bit
vmware-iso: Installing 64-bit JRE8...
vmware-iso: JRE8 has been installed.
vmware-iso: Searching if the previous version exists...
vmware-iso:   jre8 may be able to be automatically uninstalled.
vmware-iso: Environment Vars (like PATH) have changed. Close/reopen your shell to
vmware-iso:  see the changes (or in powershell/cmd.exe just type `refreshenv`).
vmware-iso:  The install of jre8 was successful.
vmware-iso:   Software installed to 'C:\Program Files\Java\jre1.8.0_261\'                 <<<<<<<<
vmware-iso:
vmware-iso: javaruntime-platformspecific v7.0.79.20161125 [Approved]
vmware-iso: javaruntime-platformspecific package files install completed. Performing other installation steps.
```
When I booted up the vm and check the dir this was not the case, jre1.8.0_261  was not installed as logged in the output:
```
Directory of c:\Program Files\Java
07/19/2020  03:10 AM    <DIR>          .
07/19/2020  03:10 AM    <DIR>          ..
07/19/2020  03:11 AM    <DIR>          jdk1.8.0_211
07/19/2020  03:09 AM    <DIR>          jre1.8.0_251
               0 File(s)              0 bytes
               4 Dir(s)  45,883,371,520 bytes free
```
And manually running the **asadmin.bat create-service** command fails everytime:
```
C:\Users\vagrant>cmd.exe /c "C:\glassfish\glassfish4\bin\asadmin.bat create-service domain1"
java.io.IOException
Command create-service failed.
```

I manually installed the **jre1.8.0_261** package:
```
c:\Program Files\Java>dir
 Volume in drive C is Windows 2008R2
 Volume Serial Number is 4CBD-769E

 Directory of c:\Program Files\Java

08/16/2020  11:06 AM    <DIR>          .
08/16/2020  11:06 AM    <DIR>          ..
07/19/2020  03:11 AM    <DIR>          jdk1.8.0_211
07/19/2020  03:09 AM    <DIR>          jre1.8.0_251
08/16/2020  11:06 AM    <DIR>          jre1.8.0_261
               0 File(s)              0 bytes
               5 Dir(s)  44,708,659,200 bytes free
```
But still the issue persist with the same Java IOException. 
I played around the with the glassfish asadmin.bat file, setting the java references manually, but nothing I did worked. Still the same error every time
```
java.io.IOException
Command create-service failed.
```
My environment is on VMFusion. 
At least the good news is that my metasploitable3-win2k8 is up and running, nontheless.
Happy to know if anyone finds the reason this is failing for this particular install instance.

**Comment by jmartin-r7** [comment]
Fixed by #478
